reproducible-build/Dockerfile

```diff
@@ -1,7 +1,7 @@
-FROM crops/poky@sha256:f51ae3279f98768514273061336421e686e13d0a42fdb056c0b88c9afeec8c56 as builder
+FROM docker.io/crops/poky@sha256:f51ae3279f98768514273061336421e686e13d0a42fdb056c0b88c9afeec8c56 as builder
 
 USER root
-RUN apt install -y repo
+RUN apt update && apt install -y repo
```

Comment:
We need to substantially refactor the Docker image to achieve reproducibility. Point it to the snapshot and pin the versions of the installed packages. In the current version it is both not reproducible and will break once APT cache becomes too old and APT will break complaining about it.

Reply:
can you elaborate more on why is the current docker image is not reproducible? 
It is pinning the version of the yocto image to build stuff with it and installing only the avaliable repo in that image hash without pulling the newest changes in the APT cache (as this would be different in separate builds and might cause different measurements).
Is it really that important to make apt update here?
To what repo concerns and package versions, we are going to pin them in the manifest too to ensure the consistency and reproducibility.

cc @Ruteri

Reply:
Removed `apt update` to unblock us to move forward and decide later about the better approach.